### PR TITLE
bugfix - A collection of tiny tweaks to get ayon running with maya

### PR DESCRIPTION
**Pull request by sjt-rvx**
This is a collection of small tweaks to get maya (and ayon in general) to not error so I can test it. Most of this is names of things have changed just slightly.

### CODE REVIEW

File: openpype/client/server/conversion_utils.py [hunk 1]
```diff
@@ -584,7 +584,9 @@ def representation_fields_v3_to_v4(fields, con):
     representation_attributes = con.get_attributes_for_type("representation")
 
     output = set()
-    for field in fields:
+    fields = set(fields)
+    iterfields = fields.copy()
+    for field in iterfields:
```
Comment (iLLiCiTiT) [inline openpype/client/server/conversion_utils.py:589]: Why is this needed?

File: openpype/hosts/maya/api/lib.py [hunk 2]
```diff
@@ -3981,7 +3981,7 @@ def create_rig_animation_instance(nodes, context, namespace, log=None):
     )
     assert roots, "No root nodes in rig, this is a bug."
 
-    asset = legacy_io.Session["AVALON_ASSET"]
+    asset = get_current_asset_name()  # remove legacy_io call
```
Comment (iLLiCiTiT) [inline openpype/hosts/maya/api/lib.py:3984]:
```suggestion
    asset = get_current_asset_name()
```

File: openpype/hosts/maya/api/lib.py [hunk 3]
```diff
@@ -3928,7 +3928,7 @@ def get_capture_preset(task_name, task_type, subset, project_settings, log):
     }
 
     plugin_settings = project_settings["maya"]["publish"]["ExtractPlayblast"]
-    if plugin_settings["profiles"]:
+    if plugin_settings.get("profiles", []):
```
Comment (iLLiCiTiT) [inline openpype/hosts/maya/api/lib.py:3931]: Please don't make backwards compatibility because of AYON. AYON is not production ready and crashes because of inconsistency is only way how to find out we have something off...
```suggestion
    if plugin_settings["profiles"]:
```

File: openpype/lib/applications.py [hunk 4]
```diff
@@ -1516,7 +1516,7 @@ def prepare_app_environments(
     tool_by_group_name = collections.defaultdict(dict)
     if asset_doc:
         # Make sure each tool group can be added only once
-        for key in asset_doc["data"].get("tools_env") or []:
+        for key in asset_doc["data"].get("tools") or []:
```
Comment (iLLiCiTiT) [inline openpype/lib/applications.py:1519]: This would break openpype, can't be used, we have to fix it other way.

File: openpype/hosts/maya/api/lib.py (hunk 3 above)
Comment (iLLiCiTiT) [inline openpype/hosts/maya/api/lib.py:3931]: Please don't make backwards compatibility because of AYON. AYON is not production ready and crashes because of inconsistency is only way how to find out we have something off...
```suggestion
    if plugin_settings["profiles"]:
```

File: openpype/client/server/conversion_utils.py (hunk 1 above)
Comment (BigRoy) [reply openpype/client/server/conversion_utils.py:589]: I believe it was to resolve this bug reported here: https://github.com/ynput/OpenPype/issues/4949#issue-1701076658 which fails on [this line](https://github.com/ynput/OpenPype/blob/release/next-minor/openpype/client/server/conversion_utils.py#L603-L606) due to `fields` not being a `set`. 

A copy is likely made since it appears that the `fields` variable is being update as it is being iterated over. So it should basically be this:
```python
fields = set(fields)
for field in fields.copy():
    ...
```

Comment (iLLiCiTiT) [reply openpype/client/server/conversion_utils.py:589]: Isn't the bug that there should be `output |= ...` instead of copying source fields?

File: openpype/hosts/maya/api/lib.py (hunk 3 above)
Comment (tokejepsen) [inline openpype/hosts/maya/api/lib.py:3931]:
```suggestion
    if plugin_settings["profiles"]:
```

There is a fix coming for the addon settings [here](https://github.com/ynput/ayon-addons/pull/10)

File: openpype/hosts/maya/api/lib.py (hunk 2 above)
Comment (tokejepsen) [inline openpype/hosts/maya/api/lib.py:3984]:
```suggestion
    asset = get_current_asset_name()
```

Dont think we need to comment here :)

**Review by tokejepsen — CHANGES_REQUESTED** [review]

File: openpype/client/server/conversion_utils.py (hunk 1 above)
Comment (iLLiCiTiT) [inline openpype/client/server/conversion_utils.py:589]: @sjt-rvx please change this back
```suggestion
    for field in iterfields:
```

 and fix line 605 to:
```python
            output |= {
```

**Comment by sjt-rvx** [comment]
Ok, sure thing. 

I didn't realize that my repo had disabled push; first time for me dealing with github pull requests.
